src/drivers/sendgrid.ts

```
@@ -26,13 +27,45 @@ export class SGService {
   async sendEmail(mailinfo: {
       name: string;
       username: string;
-      email: string;
+      contact: string;
       action: MAIL_DEFAULTS.TEMPLATES,
+      template?: string,
+      data?: any, 
       ota?: string;
-  }[]): Promise<void> {
-        // TODO: Build the mail from mailinfo
-        const email = {};
-            await this.sgclient.send(email);
+  }): Promise<void> {
+        
+    const email = {
+      personalizations :[
+        {
+          to:[
+            {
+              email: mailinfo.contact,
+              name: mailinfo.name
+            }
+          ]
+        }
+      ],
+      content:[
+        {
+          type: "text/html", 
+          value: mailinfo.action
+        }
+      ],
+      from: {
+        email: "paigezaleppa@gmail.com",
+        name:"Inso API"
+      },
+      reply_to: {
+        email:"paigezaleppa@gmail.com",
+        name:"Inso API"
+      },
+      templateId: mailinfo.template,
+      dynamicTemplateData: mailinfo.data
+    };
+
+    await this.sgclient.send(email).catch((error) => {
+      console.log(error);
```

This should throw an Http Error.